crates/common/sync/src/lib.rs

```diff
@@ -0,0 +1,148 @@
+    fetch_blobs(
+        &checkpoint_sync_url,
+        db.clone(),
+        block.data.message.block_hash(),
+    )
+    .await?;
+
+    let state = fetch_finalized_state(&checkpoint_sync_url, block.data.message.slot).await?;
+    info!("Received block from slot:{}", slot);
+    ensure!(
+        block.data.message.slot == state.data.slot,
+        anyhow::anyhow!("Slot mismatch")
+    );
+
+    ensure!(block.data.message.state_root == state.data.state_hash());
+    let mut store = get_forkchoice_store(state.data, block.data.message, db)?;
+
+    // using sepolia genesis time until pectra mainnet
+    let time = network_spec().genesis.genesis_time + SECONDS_PER_SLOT * (slot + 1);
+    on_tick(&mut store, time)?;
+    info!("Initial Sync complete");
+
+    Ok((store, slot))
+}
+
+/// Fetch initial state from trusted RPC
+pub async fn fetch_finalized_state(
+    rpc: &str,
+    slot: u64,
+) -> anyhow::Result<BeaconVersionedResponse<BeaconState>> {
+    let body = reqwest::get(format!("{}{}{}", rpc, "/eth/v2/debug/beacon/states/", slot))
+        .await?
+        .text()
+        .await?;
+
+    let state: BeaconVersionedResponse<BeaconState> = serde_json::from_str(&body)?;
+    info!("State hash:{}", state.data.state_hash());
+
+    Ok(state)
+}
+
+/// Fetch initial block from trusted RPC
+pub async fn fetch_finalized_block(
+    rpc: &str,
+    slot: Option<u64>,
+) -> anyhow::Result<BeaconVersionedResponse<SignedBeaconBlock>> {
+    let url = match slot {
+        Some(slot) => {
+            info!("Starting from slot:{}", slot);
+            format!("{}{}{}", rpc, "/eth/v2/beacon/blocks/", slot + 1)
+        }
+        None => {
+            format!("{}{}", rpc, "/eth/v2/beacon/blocks/finalized")
+        }
+    };
+
+    let body = reqwest::get(url).await?.text().await?;
+
+    let mut block: BeaconVersionedResponse<SignedBeaconBlock> = serde_json::from_str(&body)?;
+
+    // verify slot is a start slot of epoch
+    // if not,calculate epoch start slot and fetch block at that slot
+    let epoch_start_slot =
+        compute_start_slot_at_epoch(compute_epoch_at_slot(block.data.message.slot));
+    if !epoch_start_slot == block.data.message.slot {
+        warn!("Slot {} is not start of epoch", block.data.message.slot);
+        let body = reqwest::get(format!(
+            "{}{}{}",
+            rpc, "/eth/v2/beacon/blocks/", epoch_start_slot
+        ))
+        .await?
+        .text()
+        .await?;
+        block = serde_json::from_str(&body)?;
+    }
+
+    info!("Block hash:{}", block.data.message.block_hash());
+
+    Ok(block)
+}
+
+// fetch blobs from trusted RPC
+pub async fn fetch_blobs(rpc: &str, store: ReamDB, beacon_block_root: B256) -> anyhow::Result<()> {
+    let body = reqwest::get(format!(
+        "{}{}{}",
+        rpc, "/eth/v1/beacon/blob_sidecars/", beacon_block_root
+    ))
+    .await?
+    .text()
+    .await?;
+
+    let blob_sidecar: BeaconVersionedResponse<Vec<BlobSidecar>> = serde_json::from_str(&body)?;
```

Comment:
```suggestion
    let blob_sidecar = reqwest::get(format!(
        "{rpc}/eth/v1/beacon/blob_sidecars/{}",
        rpc, beacon_block_root
    ))
    .await?
    .json::<BeaconVersionedResponse<Vec<BlobSidecar>>>()
    .await?;
```

can't we do something like this?